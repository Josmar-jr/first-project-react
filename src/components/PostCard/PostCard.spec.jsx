import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const props = {
  attributes: {
    id: 1,
    title: 'title 1',
    body: 'body 1',
    cover: 'file.png'
  }
}

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /title/i }))
      .toHaveAttribute('src', 'file.png');
    expect(screen.getByRole('heading', { name: 'title 1' }))
      .toBeInTheDocument();
    expect(screen.getByText('body 1'))
      .toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  })
})
