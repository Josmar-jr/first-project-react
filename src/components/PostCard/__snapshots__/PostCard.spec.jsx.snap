// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="sc-bdnxRM kSBbAX"
>
  <img
    alt="title 1"
    src="file.png"
  />
  <h2>
    title 1
  </h2>
  <p>
    body 1
  </p>
</div>
`;
